src/main/ruby/truffleruby/core/random.rb

```diff
@@ -66,10 +66,18 @@ def random(limit)
   end
 
   def random_range(limit)
-    min, max = limit.max.coerce(limit.min)
```

Comment:
I had to get rid of usage of Range#max here because it fails if end of the range if float